azurerm/internal/services/containers/kubernetes_cluster_resource.go

```diff
@@ -866,8 +866,12 @@ func resourceKubernetesClusterCreate(d *schema.ResourceData, meta interface{}) e
 	}
 
 	if v, ok := d.GetOk("private_dns_zone_id"); ok {
-		if (parameters.Identity == nil && !servicePrincipalSet) || (v.(string) != "System" && (!servicePrincipalSet && parameters.Identity.Type != containerservice.ResourceIdentityTypeUserAssigned)) {
-			return fmt.Errorf("a user assigned identity or a service principal must be used when using a custom private dns zone")
+		if v.(string) != "None" {
+			usingAServicePrincipal := parameters.Identity == nil && !servicePrincipalSet
+			hasAUserIdentity := !servicePrincipalSet && parameters.Identity.Type != containerservice.ResourceIdentityTypeUserAssigned
```

Comment:
Umm, wait a sec, shouldn't this be:
```suggestion
			hasAUserIdentity := !servicePrincipalSet && parameters.Identity.Type != containerservice.ResourceIdentityTypeSystemAssigned
```

Reply:
Although nevermind, I get it now :)